### PR TITLE
New Homepage

### PR DESCRIPTION
**Pull request by crhallberg**

### CODE REVIEW

File: module/GeebyDeeby/view/geeby-deeby/index/index.phtml [hunk 1]
```diff
@@ -1,39 +1,77 @@
...
+          <li><a href="<? echo $this->url('series', array('extra' => 'Comments'))?>">with comments</a></li>
+        </ul>
+        <h3>Users</h3>
+        <ul>
+          <li><a href="<? echo $this->url('users')?>">List Registered Users</a></li>
+          <li><a href="<? echo $this->url('reviews')?>">Browse Recent Reviews</a></li>
+        </ul>
+      </div>
+      <div class="col-sm-6">
+        <h3>List Items</h3>
```
Comment (demiankatz) [inline module/GeebyDeeby/view/geeby-deeby/index/index.phtml]: You shortened some headers to remove "List" but left it on others. Should we be more consistent?

File: module/GeebyDeeby/view/geeby-deeby/index/index.phtml [hunk 2]
```diff
@@ -1,39 +1,77 @@
...
+<div class="row">
+  <div class="col-md-8">
+    <h2>Welcome!</h2>
+    <p>The Edward T. LeBlanc Dime Novel Bibliography project aims to create a comprehensive online database of dime novels, story papers, reprint libraries and related materials. <a href="<? echo $this->url('about')?>">Learn More &rarr;</a></p>
+    <h2>Browse</h2>
+    <div class="row">
+      <div class="col-sm-6">
+        <h3>Series</h3>
+        <ul>
+          <li><a href="<? echo $this->url('series')?>">by name</a></li>
```
Comment (demiankatz) [inline module/GeebyDeeby/view/geeby-deeby/index/index.phtml]: Why did you change `<?=` to <? echo` here? I think `<?=` is preferred syntax.

File: module/GeebyDeeby/view/layout/layout.phtml [hunk 3]
```diff
@@ -1,23 +1,21 @@
-<? echo $this->doctype(); ?>
+<?=$this->doctype(); ?>
```
Comment (demiankatz) [inline module/GeebyDeeby/view/layout/layout.phtml]: You seem to have changed `<? echo` to `<?=` throughout this file. Not sure why that is part of this pull request. If we want to make this change (which probably makes sense) it would be better to make that change in master -- it's not specific to the dimenovels.org branch.

File: module/GeebyDeeby/view/partials/searchbox.phtml [hunk 4]
```diff
@@ -3,7 +3,7 @@
     Search:
     <input type="text" name="SearchQuery" value="<?=isset($this->layout()->query) ? $this->escapeHtml($this->layout()->query) : ''?>" size="35" maxlength="50" />
     <select name="SearchType">
-      <? foreach (array('Title', 'Person', 'Keyword', 'ISBN') as $current): ?>
+      <? foreach (['Title', 'Person', 'Keyword', 'ISBN'] as $current): ?>
```
Comment (demiankatz) [inline module/GeebyDeeby/view/partials/searchbox.phtml]: Change to short array syntax is something else that really belongs in master, not this PR.

File: module/GeebyDeebyLocal/src/GeebyDeebyLocal/Controller/SearchController.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * @category GeebyDeeby
+ * @package  Controller
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://github.com/demiankatz/Geeby-Deeby Main Site
+ */
+class SearchController extends \GeebyDeeby\SearchController
+{
+    public function homeAction()
```
Comment (demiankatz) [inline module/GeebyDeebyLocal/src/GeebyDeebyLocal/Controller/SearchController.php]: What is the purpose of this custom controller?

Comment (crhallberg) [inline module/GeebyDeebyLocal/src/GeebyDeebyLocal/Controller/SearchController.php]: To pass the podcast episodes to the home page (below)

**Comment by crhallberg** [comment]
I reduced the diffs and did my best to revert the codestyle changes. These were mostly attempts to get it working on my computer. My apologies.

File: module/GeebyDeebyLocal/src/GeebyDeebyLocal/Controller/SearchController.php (hunk 5 above)
Comment (demiankatz) [inline module/GeebyDeebyLocal/src/GeebyDeebyLocal/Controller/SearchController.php]: Unless I'm mistaken, though, the home page is IndexController/indexAction, not SearchController/homeAction.

Comment (crhallberg) [inline module/GeebyDeebyLocal/src/GeebyDeebyLocal/Controller/SearchController.php]: Must have been my inner VuFind talking. ;)

Comment (demiankatz) [inline module/GeebyDeebyLocal/src/GeebyDeebyLocal/Controller/SearchController.php]: Ha! Mind doing some troubleshooting when you get a chance and seeing if you can actually get this feature working? I'm guessing that right now it must not be displaying. If you need any real data to test with, let me know and I can help.

**Comment by crhallberg** [comment]
I think I've finally massaged this into a truer presentation. A lot of the previous setup was me trying hard to get it to work at all. Ready for another full review.

**Comment by demiankatz** [comment]
Thanks for all the work on this -- the diffs are much cleaner now and it looks great.

Before I merge, though, I'd like to think a little more about the widget piece -- it seems to make the whole page rather laggy, and I'd also want to select a better, more representative set of images if we were to go ahead with this approach.

Have you tried just using the standard Twitter user timeline widget here? I wonder if that might be a simpler thing to do, that would automatically give us the "news feed" type of functionality that I'd really like to have. If you've already tried it and found it unworkable, though, we can continue discussing the Widgetic route....

**Comment by crhallberg** [comment]
It's not unworkable, it's just generally not as visual. https://dev.twitter.com/web/embedded-timelines

The grid timeline is restricted to collections and does not respond very well. That's what I tried first.

**Comment by demiankatz** [comment]
I actually think the basic user timeline may be ideal -- it appears to respond well, it includes images as they are posted, and it will allow us to highlight our very latest project instantly on the homepage without any extra work. Obviously Twitter is not the most beautiful thing in the world, but if you can live with it, I think the functionality will be good.

**Comment by crhallberg** [comment]
No problem! Updated.

**Comment by demiankatz** [comment]
I like it -- thanks!
